doc/ss_logger.rst

```diff
@@ -0,0 +1,144 @@
+
+      logger = get_logger("example_logger")
+
+To demonstrate the full functionality of of the SmartSim logger, we include all log
+functions in the Python driver script with log messages:
+
+.. code-block:: python
+
+      logger.info("This is a message")
+      logger.debug("This is a debug message")
+      logger.error("This is an error message")
+      logger.warning("This is a warning message")
+
+Before executing the Python script, we must set the environment variable, `SMARTSIM_LOG_LEVEL`,
+to a log level. The default is `info`.
+For the example, set `SMARTSIM_LOG_LEVEL` to `quiet` in the terminal::
+    export SMARTSIM_LOG_LEVEL=quiet
+
+When we execute the script with `SMARTSIM_LOG_LEVEL=quiet`,
+the following messages from will print from above::
+    21:07:40 osprey.us.cray.com SmartSim[10950] ERROR This is an error message
+    21:07:40 osprey.us.cray.com SmartSim[10950] WARNING This is a warning message
+
+Notice that the `info` and `debug` messages were ignored. This is because by setting
+a lower severity level (`quiet`), we instruct SmartSim to ignore the higher severity levels (`debug` and `info`).
+
+Next, set `SMARTSIM_LOG_LEVEL` to `info` in the terminal::
+    export SMARTSIM_LOG_LEVEL=info
+
+When we run the script with `SMARTSIM_LOG_LEVEL=info`,
+the following output appears in stdout::
+    19:52:05 osprey.us.cray.com SmartSim[130033] INFO This is a message
+    19:52:05 osprey.us.cray.com SmartSim[130033] ERROR This is an error message
+    19:52:05 osprey.us.cray.com SmartSim[130033] WARNING This is a warning message
+
+Above, we have instructed SmartSim to print all `info` messages but ignore higher
+verbose messages (`debug`).
+
+Now set the `SMARTSIM_LOG_LEVEL` to `debug` and check the output of the program.
+Set the environment variable in the terminal like so::
+    export SMARTSIM_LOG_LEVEL=debug
+
+When we run the program once again, the following output is printed to stdout::
+    20:11:12 osprey.us.cray.com SmartSim[65385] INFO This is a message
+    20:11:12 osprey.us.cray.com SmartSim[65385] DEBUG This is a debug message
+    20:11:12 osprey.us.cray.com SmartSim[65385] ERROR This is an error message
+    20:11:12 osprey.us.cray.com SmartSim[65385] WARNING This is a warning message
+
+Notice that all log messages are visible since we set `SMARTSIM_LOG_LEVEL`
+to the highest severity log level.
+
+========================
+Example: Logging to File
+========================
+The ``log_to_file()`` function in SmartSim allows users to log messages
+to a specified file by providing a `name` to the function. The severity
+level of messages printed to the file is determined by the
+`SMARTSIM_LOG_LEVEL` variable.
+
+Begin by importing the function `get_logger` and `log_to_file`:
+
+.. code-block:: python
+
+      from smartsim.log import get_logger, log_to_file
+
+Initialize a logger for use within the Python driver script:
+
+.. code-block:: python
+
+      logger = get_logger("example_logger")
```

Comment:
Named loggers are actually ignored so "example_logger" ends up unintentionally confusing. 

Notice that the outputs show `...SmartSim[10950]...` - that's the logger name being overwritten by `get_logger`. It might be less problematic to just replace examples so the names supplied to `get_logger` are consistent w/outputs, e.g. `get_logger("SmartSim")`

Reply:
ahhh I see